src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,7 +32,14 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        // a/BLOCK, STREET, UNIT, POSTAL_CODE
+        String[] components = address.split(",");
+        Block block = new Block(components[BLOCK].substring(2, components[BLOCK].length()).trim());
+        Street street = new Street(components[STREET].trim());
+        Unit unit = new Unit(components[UNIT].trim());
+        PostalCode postal = new PostalCode(components[POSTAL].trim());
         this.value = address;
+        
```

Comment:
why is `Address.value` still retained? you will only introduce duplicity and inconsistency by retaining this variable.